xbatcher/accessors.py

```diff
@@ -1,8 +1,21 @@
+from typing import Union
+
 import xarray as xr
 
 from .generators import BatchGenerator
 
 
+def _as_xarray_dataarray(xr_obj: Union[xr.Dataset, xr.DataArray]) -> xr.DataArray:
+    """
+    Convert xarray.Dataset to xarray.DataArray if needed, so that it can
+    be converted into a torch.Tensor object.
```

Comment:
Not just for `torch.Tensor` anymore.

```suggestion
    be converted into a Tensor object.
```